main.py

```diff
@@ -427,6 +429,63 @@ def update():
     conn.close()
     return ''
 
+@app.route('/add_run3', methods=['POST'])
+def add_run3():
+    """
+    Endpoint to add a free text sample in run3 planning sheet
+    """
+    conn = sqlite3.connect('run3.db')
+    cursor = conn.cursor()
+    user_info = get_user_info(cursor)
+    if user_info['role'] == 'not a user':
+        logging.error('Could not find user %s, not doing anything', user_info)
+        return 'You are not a user of McM', 403
+
+    data = json.loads(request.data)
+
+    dataset_name = data['datasetname']
+    number_events = data['numberofevents']
+
+    pwg_list = []
+
+    sql_query = '''SELECT dataset
+                   FROM run3_samples
+                   WHERE  = ?'''
+
+    rows = cursor.execute(sql_query, [dataset_name])
+
+    #is the sample already in the list?
+    if rows is not None:
+        return ''
+
+    #input checks
+    if dataset_name is None or not number_events.replace(' ', '').isdigit():
+        return ''
+
+    #pwg checks: table is updated if there is at least 1 valid pwg
+    if 'pwginterested' in data:
```

Comment:
I think this `if` and `for` can be slightly changed:
```
# Get pwginterested or empty string, uppercase it and split on commas
for pwg in data.get('pwginterested', '').upper().split(','):
    # Remove any surrounding whitespaces, if any
    pwg = pwg.strip()
    if not pwg:
        # If nothing is left after strip, continue
        continue

    if pwg not in all_pwgs:
        # If given PWG is not a valid one
        return 'Bad PWG %s' % (pwg), 400

    # Add PWG to a list
    pwg_list.append(pwg)
```
Idea is the same, just avoiding `if 'pwginterested'` and doing `.strip()` on each element instead of space-replacement once in the beginning.